Update Publishing your website article (#37958)
mdn/content: files/en-us/learn_web_development/getting_started/your_first_website/publishing_your_website/index.md

- Web hosting is rented file space on a hosting company's [web server](/en-US/docs/Learn_web_development/Howto/Web_mechanics/What_is_a_web_server). You put website files on the web server. The web server provides website content to website visitors.
- A [domain name](/en-US/docs/Learn_web_development/Howto/Web_mechanics/What_is_a_domain_name) is the unique web address where people find your website, such as `https://www.mozilla.org` or `https://www.bbc.co.uk`. You can rent your domain name for as many years as you want from a **domain registrar**.

If you get your web hosting _and_ domain name from the same company, they tend to be configured automatically to talk to one another. However, If you get them from separate companies, or want to change your hosting to a different company, you need to do a bit of setup to point the domain name to the correct server. This is so that people will see your website when they navigate to that web address. This is usually done by logging into your domain registrar's website, and setting your domain's [nameservers](https://kinsta.com/knowledgebase/what-is-a-nameserver/) to the ones provided by your hosting company.

> To actually transfer the website files on to the hosting company's web server, different companies will have different mechanisms available. Many will have more than one option; typical options include:

Reviewer: ```suggestion
Companies use various mechanisms to transfer files to their web servers. Many will have more than one option; typical options include:
```

- A drag and drop interface (you'll see an example of this in [Publishing via GitHub](#publishing_via_github), later on).
- An {{Glossary("FTP", "File Transfer Protocol (FTP)")}} program. FTP programs vary widely, but generally, you have to connect to your web server using details provided by your hosting company (typically username, password, hostname). Then the program shows you your local files and the web server's files in two windows, and provides a way for you to transfer files back and forth.

> - Keeping your source code in a GitHub repo (again, see below) and granting the hosting comppany access to that repo so they can grab the code from there to publish.

Reviewer: ```suggestion
- Keeping website source code in a GitHub repo (see below) and granting the hosting company access so they can fetch the source, build it if necessary, and publish it.
```

2. Next, you need to [create a repository](https://github.com/new) to store files. On this page:
   1. in the _Repository name_ box, enter _username_.github.io, where _username_ is your username. For example, our friend Bob Smith would enter _bobsmith.github.io_.
   2. Click the _Create repository_ button at the bottom of the page.
3. On the next page, find the _uploading an existing file_ link, and click on it. This should bring you to the file upload page.
4. At this point, you should be able to drag and drop files from your local file system onto the web page to upload them to the GitHub repository. To do so:
   1. Open a file explorer/finder window on your computer.
   2. Make sure you can see the file explorer _and_ the web browser windows — position them next to one another on your screen.
   3. Navigate the file explorer window to the folder containing your sample website.
      > [!NOTE]
      > Make sure your folder has an `index.html` file.
   4. Select all of your sample website's files (for example using the <kbd>Ctrl</kbd> + <kbd>A</kbd> keyboard shortcut, or <kbd>Cmd</kbd> + <kbd>A</kbd> on macOS).

>    5. With your mouse pointer over the selected files, hold down your left mouse button and drag then from your file explorer over the "Drag files here to add them to your repository" section of the GitHub tab in your web browser window.

Reviewer: ```suggestion
   5. Drag the files from your file explorer over the "Drag files here to add them to your repository" section of the GitHub page.
```

>    6. The section's border and text should change to indicate when you are ready to drop the files. Release the mouse button when this happens.

Reviewer: ```suggestion
   6. The section's border and text changes to indicate a drop is possible. Drop the files at this point.
```
